sprocs/sync_assessments.sql

```diff
@@ -414,6 +416,14 @@ BEGIN
         AND a.deleted_at IS NULL
         AND a.course_instance_id = syncing_course_instance_id
         AND (da.errors IS NOT NULL AND da.errors != '');
+    
+    -- Ensure all assessments have an assessment unit, default number=0.
+    UPDATE assessments AS a
+    SET
+        assessment_unit_id = COALESCE(a.assessment_unit_id,
+            (SELECT id FROM assessment_units WHERE number = 0 AND course_id = syncing_course_id))
+    WHERE a.deleted_at IS NULL
+    AND a.course_instance_id = syncing_course_instance_id;
```

Comment:
I believe this is cleaner and faster, but I haven't tested it:

```suggestion
    UPDATE assessments AS a
    SET
        assessment_unit_id = 
            (SELECT id FROM assessment_units WHERE number = 0 AND course_id = syncing_course_id)
    WHERE a.deleted_at IS NULL
    AND a.course_instance_id = syncing_course_instance_id
    AND a.assessment_unit_id IS NULL;
```

Reply:
I'm not sure why, but a couple of assessments cause an error on sync when I use your suggestion.

Reply:
Did you find any particular feature in common between these assessments (e.g., they are assessments with a unit, without a unit, with a unit listed in the course.json, or not, etc.)?

Reply:
I see now that a couple of new assessments have been added to exampleCourse, obv. without units. The error is being thrown from line 472, presumably because they're making it to the end of the sync procedure with a null assessment_unit_id. Not sure why your code wouldn't work though.

On that note, I think I should come up with new units for the exampleCourse, so that when people do add new courses to it, it's more obvious what unit they should belong to.